languages/unicode.lua

```diff
@@ -66,17 +67,15 @@ SILE.nodeMakers.unicode = SILE.nodeMakers.base {
       self:makeToken()
       self:makeGlue(item)
     elseif self:isBreaking(item.text) then
-      self:addToken(char, item)
       self:makeToken()
       self:makePenalty(0)
-    elseif self.lasttype and (self.thistype and thistype ~= lasttype and not self.isWordType[thistype]) then
-      self:makeToken()
+    elseif self.lasttype and (thistype and thistype ~= self.lasttype and not self.isWordType[thistype]) then
       self:addToken(char, item)
     else
       self:letterspace()
       self:addToken(char, item)
     end
-    if not self.isWordType[thistype] then lasttype = chardata[cp] and chardata[cp].linebreak end
+    if not self.isWordType[thistype] then self.lasttype = chardata[cp] and chardata[cp].linebreak end
     self.lasttype = thistype
```

Comment:
This line will still clobber any possibility of the previous if statement doing anything at all.

Reply:
You're right. It's superfluous.